src/frontend/packages/lib_markdown/src/components/MarkdownEditor/index.tsx

```diff
@@ -144,6 +144,15 @@ export const MarkdownEditor = ({ markdownDocumentId }: MarkdownEditorProps) => {
       },
     });
 
+  useEffect(() => {
+    if (
+      markdownDocument?.translations &&
+      markdownDocument?.translations.length > 0
+    ) {
+      setLanguage(markdownDocument?.translations[0].language_code);
+    }
+  }, [markdownDocument]);
```

Comment:
To avoid unnecessary update of useEffect:
```suggestion
  useEffect(() => {
    if (markdownDocument?.translations[0]?.language_code) {
      setLanguage(markdownDocument.translations[0].language_code);
    }
  }, [markdownDocument?.translations[0]?.language_code]);
```